modules/azure_bing_search_views/src/Plugin/views/query/AzureBingSearchViews.php

```diff
@@ -0,0 +1,353 @@
+      }
+    }
+
+    return $results;
+  }
+
+  /**
+   * Get query result.
+   *
+   * @param int $n
+   *   Number of items.
+   * @param int $offset
+   *   Offset of items (0-indexed).
+   *
+   * @return object|null
+   *   Decoded response from Bing, or empty array on error.
+   */
+  protected function getResults($n = 1, $offset = 0) {
+    $params = [
+      'offset' => $offset,
+      'count' => $n,
+    ];
+
+    // Grab data regarding conditions placed on the query.
+    $query = $this->view->build_info['query'];
+
+    // TODO find a better way.
+    $keyword = $query['keyword'][0];
+
+    return $this->bingCustomSearch->searchResults($keyword, $params);
+  }
+
+  /**
+   * Adds a simple condition to the query.
+   *
+   * Collect data on the configured filter criteria so that we can
+   * appropriately apply it in the query() and execute() methods.
+   *
+   * @param $group
+   *   The WHERE group to add these to; groups are used to create AND/OR
+   *   sections. Groups cannot be nested. Use 0 as the default group.
+   *   If the group does not yet exist it will be created as an AND group.
+   * @param $field
+   *   The name of the field to check.
+   * @param $value
+   *   The value to test the field against. In most cases, this is a scalar.
+   *   For more complex options, it is an array.
+   *   The meaning of each element in the array is dependent on the $operator.
+   * @param $operator
+   *   The comparison operator, such as =, <, or >=. It also accepts more
+   *   complex options such as IN, LIKE, LIKE BINARY, or BETWEEN. Defaults to =.
+   *   If $field is a string you have to use 'formula' here.
+   *
+   * @see \Drupal\Core\Database\Query\ConditionInterface::condition()
+   * @see \Drupal\Core\Database\Query\Condition
+   */
+  public function addWhere($group, $field, $value = NULL, $operator = NULL) {
+    // Ensure all variants of 0 are actually 0. Thus '', 0 and NULL are all
+    // the default group.
+    if (empty($group)) {
+      $group = 0;
+    }
+
+    // Check for a group.
+    if (!isset($this->where[$group])) {
+      $this->setWhereGroup('AND', $group);
+    }
+
+    $this->where[$group]['conditions'][] = [
+      'field' => $field,
+      'value' => $value,
+      'operator' => $operator,
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function defineOptions() {
+    $options = parent::defineOptions();
+    $options['safeSearch'] = [
+      'default' => 'Off',
+    ];
+
+    return $options;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildOptionsForm(&$form, FormStateInterface $form_state) {
+    parent::buildOptionsForm($form, $form_state);
+
+    // See https://docs.microsoft.com/en-us/rest/api/cognitiveservices-bingsearch/bing-custom-search-api-v7-reference#query-parameters
+    $form['safeSearch'] = [
+      '#title' => $this->t('Safe Search'),
+      '#type' => 'select',
+      '#options' => [
+        'Off' => t('Off'),
+        'Moderate' => t('Moderate'),
```

Comment:
Probably better to use $this->t() instead of t() in classes

Reply:
resolved